File: core/src/com/ultreon/craft/registry/Registries.java
package com.ultreon.craft.registry;

import com.ultreon.craft.UltreonCraft;
import com.ultreon.craft.block.Block;
import com.ultreon.craft.world.gen.noise.NoiseSettings;
import com.ultreon.libs.registries.v0.Registry;

public final class Registries {
    public static final Registry<Block> BLOCK = Registry.create(UltreonCraft.id("block"));
    public static final Registry<NoiseSettings> NOISE_SETTINGS = Registry.create(UltreonCraft.id("noise_settings"));
    public static final Registry<Registry<?>> REGISTRY = Registry.create(UltreonCraft.id("registry"));

    public static void init() {

    }
}

File: src/jvmMain/kotlin/com/ultreon/craft/registry/Registries.kt
package com.ultreon.craft.registry

import com.ultreon.craft.*
import com.ultreon.craft.resources.*
import kotlin.reflect.*

object Registries {
    val noiseSettings = registry<NoiseSettings>()
    val registries = registry<Registry<*>>()

    fun <T : Any> getRegistry(kClass: KClass<T>): Registry<T> {
        return Registry[kClass]
    }

}
